### PR TITLE
ISPN-11704 Change from FileSystemBind to Docker Volume in Server Test Suite

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-11704

### CODE REVIEW

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java [hunk 1]
```diff
@@ -155,41 +152,10 @@ protected void start(String name, File rootDir, String configurationFile) {
...
-                     .run("/bin/sh", "-c", "if [ -x \"$(command -v usermod)\" ]; then usermod -u " + uid +" jboss; fi")
-                     .run("/bin/sh", "-c", "if [ -x \"$(command -v groupmod)\" ]; then groupmod -g " + gid + " jboss; fi");
-            }
-            builder
-                  .copy("build", INFINISPAN_SERVER_HOME)
-                  .run("chown", "-R", "jboss:jboss", INFINISPAN_SERVER_HOME)
-                  .user("jboss");
-         }
-         // Copy the resources to a location from where they can be added to the image
-         try {
```
Comment (diegolovison) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: @tristantarrant I didn't find a use case for this? Could you please help me on that ?

**Comment by diegolovison** [comment]
I am going to investigate the failures

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java (hunk 1 above)
Comment (tristantarrant) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: This is needed when using this out-of-tree. It must be kept

**Comment by diegolovison** [comment]
There are 4 errors in `org.infinispan.server` that I don't see how they can be related to this PR:
- persistence.ManagedConnectionOperations.testPutGetRemove
- persistence.ManagedConnectionOperations.testTwoCachesSameCacheStore
- hotrod.HotRodMergeTest.createBeforeClass
- hotrod.HotRodMergeTest.testNewTopologySentAfterOverlappingMerge[DIST_SYNC, ALLOW_READ_WRITES]

For the 4 errors mentioned above can we fix in a different PR?

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java (hunk 1 above)
Comment (diegolovison) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: What means `using this out-of-tree`? What should be the test that must fail if we don't have this code?

Comment (diegolovison) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: I have added the code

Comment (tristantarrant) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: Can you please try running your changes with https://github.com/tristantarrant/infinispan-playground-testdriver ?

Comment (diegolovison) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: I was not able to create a cluster until https://github.com/tristantarrant/infinispan-playground-testdriver/pull/3

I have fixed `ContainerInfinispanServerDriver`
The main changes were:
* `public static final String IMAGE_USER = "185";`
* `builder.run("usermod -u " + IMAGE_USER + " jboss");`
The prebuilt image has the user 185, in this case, now we are also setting the user `185`

Comment (tristantarrant) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: I'd prefer doing this dynamically as there is no guarantee that the user id will stay constant and might also change depending on upstream/downstream/version/etc

Comment (diegolovison) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java:177]: I have changed the user to `200` and also removed:
```
if (!prebuiltImage) {
  builder.run("usermod -u " + IMAGE_USER + " jboss");
}
```
It means that, if we built the image on the fly, will have a user `200`
If we use a prebuilt image, we are setting the user to `200`

When executing `chown` or `chmod` with an UID, Linux won't fail. ( This was a surprise for me)

I have executed the test in https://github.com/tristantarrant/infinispan-playground-testdriver and it passed.

**Comment by tristantarrant** [comment]
Merged and backported to 10.1. Thanks

**Pull request merged**